src/network.cpp

```diff
@@ -64,13 +66,59 @@ namespace oxen::quic
 
     void Network::close()
     {
-        quic_manager->close_all();
-        std::this_thread::sleep_for(std::chrono::milliseconds(2500));
-    }
+        if (not running.exchange(false))
+            return;
 
-    void Network::run()
-    {
-        ev_loop->run();
+        log::info(log_cat, "Shutting down context...");
+
+        std::promise<void> p;
+        auto f = p.get_future();
+        quic_manager->call([this, &p]() {
+            try
+            {
+                quic_manager->close_all();
+
+                for (auto& [addr, handle] : mapped_client_addrs)
+                {
+                    auto s = handle->sock();
+                    log::debug(log_cat, "closing client UDPHandle bound to {}:{}", s.ip, s.port);
+                    handle->close();
+                }
+                for (auto& [addr, handle] : mapped_server_addrs)
+                {
+                    auto s = handle->sock();
+                    log::debug(log_cat, "closing server UDPHandle bound to {}:{}", s.ip, s.port);
+                    handle->close();
+                }
+                if (loop_thread)
+                {
+                    ev_loop->walk(uvw::Overloaded{
+                            [](uvw::UDPHandle&& h) {
+                                auto s = h.sock();
+                                log::debug(log_cat, "closing UDPHandle bound to {}:{}", s.ip, s.port);
+                                h.close();
+                            },
+                            [](auto&&) {}});
```

Comment:
We're going to have to had overloads here for the other types of handles (async, timer, etc). 

The example uvw has in their documentation of closing all handles through a generic callback doesn't work.
[Jason](https://github.com/skypjack/uvw/issues/239) brought up this issue 2 years ago, so it's great it's still present. Hopefully this would be fixed when we bump or replace uvw/libuv

Reply:
I'm probably gonna just get rid of this call to `walk` entirely as it seems kinda jank and currently does nothing for us.

Reply:
(and then just make sure we keep track of and clean up uvw handles appropriately rather than trying to get it to automagically keep track of them)

Reply:
In fact, we kinda *need* to not rely on walk unless we can afford to use all our handles' user_data solely to show ourselves that we own the handle, because we may be given an existing uv_loop and we can't trample the user's handles.

Reply:
yeeeeup hopefully when we merge #8, we can just call
```
ev_loop->walk([](auto &&h){ h.close(); });
```